Mathlib/FieldTheory/IsAlgClosed/Basic.lean

```diff
@@ -232,36 +242,37 @@ variable {K : Type u} [Field K] {L : Type v} {M : Type w} [Field L] [Algebra K L
   closed extension of K, then any embedding of L/K into M/K extends to an embedding of E/K.
   Known as the extension lemma in https://math.stackexchange.com/a/687914. -/
 theorem surjective_comp_algebraMap_of_isAlgebraic {E : Type*}
-    [Field E] [Algebra K E] [Algebra L E] [IsScalarTower K L E] (hE : Algebra.IsAlgebraic L E) :
+    [Field E] [Algebra K E] [Algebra L E] [IsScalarTower K L E] [Algebra.IsAlgebraic L E] :
     Function.Surjective fun φ : E →ₐ[K] M ↦ φ.comp (IsScalarTower.toAlgHom K L E) :=
   fun f ↦ IntermediateField.exists_algHom_of_splits'
-    (E := E) f fun s ↦ ⟨(hE s).isIntegral, IsAlgClosed.splits_codomain _⟩
+    (E := E) f fun s ↦ ⟨Algebra.IsIntegral.isIntegral s, IsAlgClosed.splits_codomain _⟩
 
-variable (hL : Algebra.IsAlgebraic K L) (K L M)
+variable [Algebra.IsAlgebraic K L] (K L M)
 
 /-- Less general version of `lift`. -/
 private noncomputable irreducible_def lift_aux : L →ₐ[K] M :=
   Classical.choice <| IntermediateField.nonempty_algHom_of_adjoin_splits
-    (fun x _ ↦ ⟨(hL x).isIntegral, splits_codomain (minpoly K x)⟩)
+    (fun x _ ↦ ⟨Algebra.IsIntegral.isIntegral x, splits_codomain (minpoly K x)⟩)
     (IntermediateField.adjoin_univ K L)
 
 variable {R : Type u} [CommRing R]
 variable {S : Type v} [CommRing S] [IsDomain S] [Algebra R S] [Algebra R M] [NoZeroSMulDivisors R S]
-  [NoZeroSMulDivisors R M] (hS : Algebra.IsAlgebraic R S)
+  [NoZeroSMulDivisors R M] [Algebra.IsAlgebraic R S]
 
 variable {M}
 
-private theorem FractionRing.isAlgebraic :
+private instance FractionRing.isAlgebraic :
```

Comment:
Do we really want to keep this `private`?

Reply:
I think so: since we have to declare the `Algebra` instance inline I don't see how this will apply often or at all.

Reply:
In that case, shouldn't it be local?

Reply:
So you mean `private theorem FractionRing.isAlgebraic` followed by `attribute [local instance] FractionRing.isAlgebraic`?

Reply:
Aah, then the diff that Github shows me at the top of this conversation is misleading. Because it suggests taht tit is a `private instance`.